diff --git a/extensions/github-authentication/src/common/env.ts b/extensions/github-authentication/src/common/env.ts
index 18fd7327b88..24f1c79cda8 100644
--- a/extensions/github-authentication/src/common/env.ts
+++ b/extensions/github-authentication/src/common/env.ts
@@ -5,8 +5,20 @@
 import { Uri } from 'vscode';
 import { AuthProviderType } from '../github';
 
-export function isSupportedClient(_uri: Uri): boolean {
-	return false;
+const VALID_DESKTOP_CALLBACK_SCHEMES = [
+	'macide',
+	'vscode',
+	'vscode-insiders',
+	'vscode-wsl',
+	'vscode-exploration'
+];
+
+export function isSupportedClient(uri: Uri): boolean {
+	return (
+		VALID_DESKTOP_CALLBACK_SCHEMES.includes(uri.scheme) ||
+		/(?:^|\.)vscode\.dev$/.test(uri.authority) ||
+		/(?:^|\.)github\.dev$/.test(uri.authority)
+	);
 }
 
 export function isSupportedTarget(type: AuthProviderType, gheUri?: Uri): boolean {
@@ -16,6 +28,6 @@ export function isSupportedTarget(type: AuthProviderType, gheUri?: Uri): boolean
 	);
 }
 
-export function isHostedGitHubEnterprise(_uri: Uri): boolean {
-	return false;
+export function isHostedGitHubEnterprise(uri: Uri): boolean {
+	return /\.ghe\.com$/.test(uri.authority);
 }
